controllers/front/listing/CategoryController.php

```diff
@@ -48,24 +48,29 @@ public function canonicalRedirection($canonicalURL = '')
         }
     }
 
-    public function getCanonicalURL()
+    /**
+     * {@inheritdoc}
+     *
+     * @return string
+     */
+    public function getCanonicalURL(): string
```

Comment:
```suggestion
    public function getCanonicalURL()
```

Reply:
That avoids the BC Break

Reply:
Hello @Progi1984 

Thanks for the review. 
For my personal culture, can you tell me how it creates a BC Break then on ProductController there is the same thing? 

Thanks a lot!

Reply:
Well, it's usually a good thing to type hint the return type of a method.

But in the case of an open-source project dealing with retro-compatibility, adding this later on is a problem, for example, if a module overrides this method, then with your modification, his override will not be compatible anymore, it will trigger a fatal error.

Reply:
Okay, I see, no problem and thanks for the explanation :)!